src/labone/mock/hpk_mock_server.py

```diff
@@ -0,0 +1,137 @@
+SERVER_ERROR = "SERVER_ERROR"
+
+
+
+def build_capnp_error(error: Exception):
+    """
+    Returns:
+        Capnp Type dictionary for: Result(Error)
+    """
+    return {
+        "err": {
+            "code": 2,
+            "message": f"{error}",
+            "category": SERVER_ERROR,
+            "source": __name__,
+        }
+    }
+
+
+class HpkMockServer(ConcreteMockServer):
+    id = HPK_SCHEMA_ID
+    name = "HPK"
+
+    def __init__(self, functionality: HpkMockFunctionality) -> None:
+        super().__init__()
+        self._functionality = functionality
+
+    async def getTheSchema(self, _context, **kwargs):
+        """Redirect to server method."""
+        return await self._base_server.getTheSchema(_context, **kwargs)
+
+    async def listNodes(self, pathExpression, flags, client, _context, **kwargs):
+        return await self._functionality.list_nodes(pathExpression, flags=flags)
+
+    async def listNodesJson(self, pathExpression, flags, client, _context, **kwargs):
+        return json.dumps(
+            await self._functionality.list_nodes_info(
+                path=pathExpression,
+                flags=flags,
+            )
+        )
+
+    async def getValue(
+        self,
+        pathExpression,
+        lookupMode,
+        flags,
+        client,
+        _context,
+        **kwargs,
+    ):
+        try:
+            if lookupMode == 0:  # direct lookup
+                responses = [await self._functionality.get(pathExpression)]
+            else:
+                responses = await self._functionality.get_with_expression(
+                    pathExpression, flags=flags
+                )
+        except BaseException as e:
+            return [build_capnp_error(e)]
+
+        return [
+            {
+                "ok": {
+                    "value": {"int64": response.value},
+                    "metadata": {
+                        "path": response.path,
+                        "timestamp": response.timestamp,
+                    },
+                },
+            }
+            for response in responses
+        ]
+
+    async def setValue(
+        self,
+        pathExpression,
+        value,
+        lookupMode,
+        completeWhen,
+        client,
+        _context,
+        **kwargs,
+    ):
+        value = value.int64
+        try:
+            if lookupMode == 0:  # direct lookup
+                responses = [
+                    await self._functionality.set(
+                        AnnotatedValue(value=value, path=pathExpression),
+                    ),
+                ]
+            else:
+                responses = await self._functionality.set_with_expression(
+                    AnnotatedValue(value=value, path=pathExpression),
+                )
+        except BaseException as e:
+            return [build_capnp_error(e)]
+
+        return [
+            {
+                "ok": {
+                    "value": {"int64": response.value},
```

Comment:
issue: You explicitly allow only integer nodes with this! This already exists in `labone.core.value._value_from_python_types`